fieldmask/validate.go

```diff
@@ -22,12 +22,20 @@ func Validate(fm *fieldmaskpb.FieldMask, m proto.Message) error {
 	md0 := m.ProtoReflect().Descriptor()
 	for _, path := range fm.GetPaths() {
 		md := md0
+		var fd protoreflect.FieldDescriptor
 		if !rangeFields(path, func(field string) bool {
 			// Search the field within the message.
 			if md == nil {
 				return false // not within a message
 			}
-			fd := md.Fields().ByName(protoreflect.Name(field))
+			// Parent message is a repeated field.
+			// Use of AIP161 optional wildcards to target sub-fields
+			// on repeated fields is not supported
```

Comment:
isn't the spec saying that it is supported on repeated fields, but the field in question must be a wildcard? `*`?

Reply:
> Field masks may permit the use of the * character on a repeated field or map to indicate the specification of particular sub-fields in the collection:

Reply:
It does maybe the comment should be reworded, but it felt worthy of a clarification. 
My intention was not to add the optional support for wildcard values in here but tackle the faulty interpretation of fieldmasks that don't traverse using wildcard `*` 
The choice to disallow the use of these wildcards was then to bring it inline with how the fieldmask update function behaves where list traversal is not permitted. The AIP rule is also stated as you *MAY* support it, meaning we can stick to not supporting it. 

I can look into adding the support to traverse with wildcard in both the validate and update functions if that would be something we want to add?